src/runtimes/node/bundlers/index.ts

```diff
@@ -57,15 +61,15 @@ const getDefaultBundler = async ({
   mainFile: string
   featureFlags: FeatureFlags
 }): Promise<NodeBundlerType> => {
-  if (['.mjs', '.ts'].includes(extension)) {
+  if (ESBUILD_EXTENSIONS.has(extension)) {
     return NodeBundlerType.ESBUILD
   }
 
   if (featureFlags.traceWithNft) {
     return NodeBundlerType.NFT
   }
 
-  const functionIsESM = await detectEsModule({ mainFile })
+  const functionIsESM = extname(mainFile) !== '.cjs' && (await detectEsModule({ mainFile }))
```

Comment:
I wonder if it'd make sense to have the extension check part of `detectEsModule`? Not a blocker, though.

Reply:
I did not want to mess with `detectEsModule` as it is actually detecting es modules based on the contents. The only extensions that reach this line are `cjs` and `js`. So I wasn't sure if adding the single check for `cjs` would make sense in `detectEsModule`